tool/src/main/java/org/wildfly/security/tool/ElytronTool.java

```diff
@@ -131,15 +138,26 @@ private static void configureLogManager() {
     }
 
     private void generalHelp() {
-        System.out.print(ElytronToolMessages.msg.generalHelpTitle());
-        System.out.println();
-        for (Command c: commandRegistry.values()) {
-            if (scriptName != null) {
-                c.setToolCommand(scriptName);
-            }
-            c.help();
-            System.out.println();
-        }
+        DescriptionSection descriptionSection = new DescriptionSection(ElytronToolMessages.msg.cmdElytronToolDescription());
+        UsageSection usageSection = new UsageSection(null, null);
+        OptionsSection optionsSection = new OptionsSection(ElytronToolMessages.msg.generalHelpOptionsOpening(), null);
+
+        // Using SortedMap so commands are in alphabetical order
+        SortedMap<String, String> commandsMap = new TreeMap<>();
+        commandsMap.put(CredentialStoreCommand.CREDENTIAL_STORE_COMMAND, ElytronToolMessages.msg.cmdLineCredentialStoreHelpHeader());
+        commandsMap.put(VaultCommand.VAULT_COMMAND, ElytronToolMessages.msg.cmdVaultHelpHeader());
+        commandsMap.put(FileSystemRealmCommand.FILE_SYSTEM_REALM_COMMAND, ElytronToolMessages.msg.cmdFileSystemRealmHelpHeader());
+        commandsMap.put(FileSystemEncryptRealmCommand.FILE_SYSTEM_ENCRYPT_COMMAND, ElytronToolMessages.msg.cmdFileSystemEncryptHelpHeader());
```

Comment:
We have one more new command that should also be added here and updated: `FileSystemRealmIntegrityCommand`